driver/csiplugin/controllerserver.go

```diff
@@ -608,11 +608,166 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 	if !allValid {
 		return nil, status.Error(codes.InvalidArgument, "The Parameter(s) not supported in storageClass: "+invalidParams)
 	}
-	scaleVol, err := getScaleVolumeOptions(ctx, req.GetParameters())
+
+	scaleVol, isNewVolumeType, primaryClusterID, err := cs.setScaleVolume(ctx, req, volName, volSize)
+	if err != nil {
+		return nil, err
+	}
+
+	isSnapSource, isVolSource, volSrc, snapIdMembers, srcVolumeIDMembers, err := cs.getVolORSnapMembers(ctx, req, volName)
+	if err != nil {
+		return nil, err
+	}
+
+	err = cs.checkPrimaryFileSetLink(ctx, scaleVol)
+	if err != nil {
+		return nil, err
+	}
+	if scaleVol.PrimaryFS != scaleVol.VolBackendFs {
+		// primary filesytem must be mounted on GUI node so that we can create the softlink
+		// skip if primary and volume filesystem is same
+		err := checkPrimaryFilesystemMountOnGUI(ctx, scaleVol)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	volFsInfo, err := checkVolumeFilesystemMountOnPrimary(ctx, scaleVol)
```

Comment:
Both are required as one is specifically for primary cluster. and another is applicable for any cluster. Also some additional checks are there primary, so thinking not to club together, otherwise hard to debug in case of failure.